client/src/protocolConverter.ts

```diff
@@ -257,7 +263,8 @@ export function createConverter(uriConverter?: URIConverter): Converter {
 			let result: code.MarkdownString;
 			switch (value.kind) {
 				case ls.MarkupKind.Markdown:
-					return new code.MarkdownString(value.value);
+					// https://github.com/Microsoft/vscode-languageserver-node/issues/374
+					return value.value.split(/\s*---\s*/g).map(asMarkdownString);
```

Comment:
Is this correct? What if `---` appears in a code snippet, or in the middle of a sentence?

Reply:
I am currently fixing it :-) We need to consider `***` and `___` as well and ignore html comments